logstash-core/src/main/java/org/logstash/config/ir/compiler/OutputStrategyExt.java

```diff
@@ -225,12 +224,23 @@ protected SimpleAbstractOutputStrategyExt(final Ruby runtime, final RubyClass me
             super(runtime, metaClass);
         }
 
-        @JRubyMethod(required = 4)
+        @JRubyMethod(required = 5)
         public IRubyObject initialize(final ThreadContext context, final IRubyObject[] args) {
             final RubyClass outputClass = (RubyClass) args[0];
             // Calling "new" here manually to allow mocking the ctor in RSpec Tests
-            output = args[0].callMethod(context, "new", args[3]);
+            output = args[0].callMethod(context, "new", args[4]);
             initOutputCallsite(outputClass);
+
+            final IRubyObject codecDelegatorClass = RubyUtil.RUBY.executeScript(
+                    "require 'logstash/codecs/delegator'\nLogStash::Codecs::Delegator",
+                    "");
+            // WARNING: order is important since metric= create gauges with data assigned from parent_config_reference=
+            IRubyObject codec = output.callMethod(context, "codec");
+            if (codec instanceof RubyBasicObject) {
```

Comment:
yes cold be hash in case the codec defines other parameters, that happens in a config like:
```
input {
  stdin {
   json {
     encoding => 'UTF-8'
   }
  }
}
```

Reply:
Hash is still a `RubyBasicObject` - the instanceof check should be redundant than